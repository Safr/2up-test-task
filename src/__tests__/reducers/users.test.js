import { ADD_USERS_SUCCESS } from '../../actionTypes';
import usersReducer from '../../reducers/users';

const initialState = [];

describe('Users reducer', () => {
  const user = {
    id: 1,
    name: 'name',
  };
  test('should setup default state', () => {
    const state = usersReducer(undefined, { type: '@@INIT' });
    expect(state).toMatchSnapshot();
  });
  test('should add users', () => {
    const action = { type: ADD_USERS_SUCCESS, users: [user] };
    const state = usersReducer(initialState, action);
    expect(state).toMatchSnapshot();
  });
});
