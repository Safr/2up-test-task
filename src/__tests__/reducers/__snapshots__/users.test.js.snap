// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users reducer should add users 1`] = `
Array [
  Object {
    "id": 1,
    "name": "name",
  },
]
`;

exports[`Users reducer should setup default state 1`] = `Array []`;
